eng/scripts/Language-Settings.ps1

```diff
@@ -1,6 +1,6 @@
 $Language = "c"
 $PackageRepository = "C"
-$packagePattern = "*.json"
+$packagePattern = "package-info.json"
```

Comment:
Can you help me understand, what's this change doing?

Reply:
This fix scopes release info to `package-info.json` files. We ran into an issue in C++ where a new `.json` file was introduced into the place where package release info is loaded. 

While the specific fix for C++ was for vcpkg, this narrowing of the search scope helps prevent potential future issues where a new `.json` file would be introduced. 

C++ fix: https://github.com/Azure/azure-sdk-for-cpp/pull/2330

Reply:
Thanks for the pointer, it is a similar change to https://github.com/Azure/azure-sdk-for-cpp/pull/2330